cuda_core/tests/test_linker.py

```diff
@@ -46,24 +47,23 @@ def compile_ltoir_functions(init_cuda):
     return object_code_a_ltoir, object_code_b_ltoir, object_code_c_ltoir
 
 
-culink_options = [
-    LinkerOptions(),
-    LinkerOptions(arch=ARCH, verbose=True),
-    LinkerOptions(arch=ARCH, max_register_count=32),
-    LinkerOptions(arch=ARCH, optimization_level=3),
-    LinkerOptions(arch=ARCH, debug=True),
-    LinkerOptions(arch=ARCH, lineinfo=True),
-    LinkerOptions(arch=ARCH, no_cache=True),  # TODO: consider adding cuda 12.4 to test matrix in which case this
-    # will fail. Tracked in issue #337
-]
-
-
-@pytest.mark.parametrize(
-    "options",
-    culink_options
-    if culink_backend
-    else culink_options
-    + [
+options = []
+
+if culink_backend:
+    options += [
+        LinkerOptions(),
+        LinkerOptions(arch=ARCH, verbose=True),
+        LinkerOptions(arch=ARCH, max_register_count=32),
+        LinkerOptions(arch=ARCH, optimization_level=3),
+        LinkerOptions(arch=ARCH, debug=True),
+        LinkerOptions(arch=ARCH, lineinfo=True),
+        LinkerOptions(arch=ARCH, no_cache=True),
+    ]
```

Comment:
I am very confused. Before this PR, these options are tested in both backends. Now they are tested only for cuLink. Why...?